pkg/partials/interface.go

```diff
@@ -0,0 +1,11 @@
+package partials
+
+// PartialPayload is an interface that is implemented by each of the partial
+// payload types that allows for error messages to be created using the provided
+// error and request payload, that matches the correct format required by the
+// request type. As well as for accounting the weight of the request payload,
+// which is determined by the request's method field.
+type PartialPayload interface {
+	GenerateErrorPayload(err error) ([]byte, error)
+	GetMethodWeighting() (uint64, error)
```

Comment:
I think I know what you have in mind when you say `Method` (i.e. JSON RPC), and you're thinking about tree weights, but need to take a step back and understand what we want here.

I think`GetRPCCost` or `GetRPCComputeUnits` is less implementation/protocol specific and gives us what we need.